parsers/soda/parser.py

```diff
@@ -0,0 +1,131 @@
+from ..base import Parser
+from pathlib import Path
+import pickle
+import numpy as np
+from ..datatypes import Annotation, Image, ImageAnnotationFile, Tag
+from typing import Any, List, Tuple
+from PIL import Image as PILImage
+from dataclasses import asdict
+import json
+from tqdm import tqdm
+from glob import glob
+
+
+
+class SODAParser:
+    annotations = {}
+    filename = []  
+    ALLOWED_PATHS = ["/train", "/test", "/val"]
+    
+
+    def __init__(
+        self,
+        images_dir: Path,
+        annotation_dir: Path,
+        dataset_name: str,
+        path: str = "/train",
+    ):
+        self.images_dir = images_dir
+        self.annotation_dir = annotation_dir
+        self.dataset_name = dataset_name
+
+        if path not in self.ALLOWED_PATHS:
+            raise ValueError(f"path should be one of {self.ALLOWED_PATHS}")
+
+        self.path = path
+        
+        
+        
+    def parse_annotations(self, 
+                         data_type='train',
+                         save_annotation=True)->ImageAnnotationFile:
+        
+        files = glob(f'{self.annotation_dir}/*.json')
+        
+        if data_type=='train':
+            json_data = f'{self.annotation_dir}/instance_train.json'
+        else:
+            json_data=f'{self.annotation_dir}/instance_val.json'
+            
+        f = open(json_data)
+        data = json.load(f)
+        
+        
+        for idx, file in tqdm(enumerate(data['images']), total=5000, desc='parsing'):
+            idx = idx+1 
```

Comment:
Instead of incrementing `idx` here, you can specify a starting value in `enumerate` this way
```suggestion
        for idx, file in tqdm(enumerate(data['images'], 1), total=5000, desc='parsing'):
```